src/wire/dhcpv4.rs

```diff
@@ -166,13 +362,39 @@ impl<'a> DhcpOption<'a> {
             }
             _ => {
                 skip_length = self.buffer_len();
+
+                assert!(skip_length <= buffer.len());
```

Comment:
@Dirbaio I have a problem I can't return an Error in this function, so in the case someone puts an 'Other' option with a very large payload into the function and the supplied buffer is smaller then this function fails. I don't want to silently return and ignore the option either. But the fuzzer will find this crash and stop there.

Reply:
It's OK for `emit` to panic if the supplied buffer len is not exactly equal to the length provided by `buffer_len`.It's the caller's responsibility to ensure that. 

I guess it's either a bug in the fuzzer (but from a quick look it looks OK to me) or a mismatch in the length you return from `buffer_len` and the length you require in `emit`...

Reply:
@Dirbaio Ah alright yeah the Repr should returns the incorrect size and the fuzzer allocates exactly that size, I see! As I'm looking into it that bug already existed beforehand